imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js

```diff
@@ -0,0 +1,165 @@
+  const ownerGroup = await Groups.findOne({ name: "owner" });
+  if (group._id === ownerGroup._id) {
+    throw new ReactionError("bad-request", "Cannot directly invite owner");
+  }
+
+  // An account can add another account to a group as long as the person adding
+  // has all permissions granted by that group.
+  // We can't use `userHasPermission` here because we want to make sure they
+  // have ALL the permissions rather than ANY.
+  // Accounts in the "Owner" group are able to add any user to any group,
+  // regardless of other permissions.
+  const contextUserAccount = await Accounts.findOne({ _id: userIdFromContext });
+  const isOwnerAccount = !!ownerGroup && contextUserAccount.groups.includes(ownerGroup._id);
+
+  const { permissions: groupPermissions = [] } = group;
+
+  if (!context.isInternalCall && !isOwnerAccount && difference(groupPermissions, userFromContext.roles[shopId] || []).length > 0) {
+    throw new ReactionError("access-denied", "Access Denied");
+  }
+
+  // check to see if invited user has an account
+  const invitedUser = await users.findOne({ "emails.address": email });
+
+  // if there is a userAccount, check to see if email has been verified
+  const matchingEmail = invitedUser && invitedUser.emails && invitedUser.emails.find((accountEmail) => accountEmail.address === email);
+  const isEmailVerified = matchingEmail && matchingEmail.verified;
+
+  // set variables to pass to email templates
+  const inviteeName = getCurrentUserName(contextUserAccount);
+  let dataForEmail;
+  let userId;
+  let templateName;
+
+  if (invitedUser) {
+    userId = invitedUser._id;
+  }
+
+  // if the user already has an account, send in informative email instead of an invite email
+  if (invitedUser && isEmailVerified) {
+    // make sure user has an `Account` - this should always be the case
+    const invitedAccount = await Accounts.findOne({ _id: userId });
+    if (!invitedAccount) throw new ReactionError("not-found", "User found but matching account not found");
+
+    await context.mutations.addAccountToGroup({ ...context, isInternalCall: true }, {
+      accountId: invitedAccount._id,
+      groupId
+    });
+
+    // do not send token, as no password reset is needed
+    const url = Reaction.absoluteUrl();
+
+    // use primaryShop's data (name, address etc) in email copy sent to new shop manager
+    dataForEmail = getDataForEmail({ shop: primaryShop, inviteeName, name, url });
+
+    // Get email template and subject
+    templateName = "accounts/inviteShopMember";
+  } else {
+    // TODO: this entire section uses meteor
+    // TODO: this entire section uses meteor
+    // TODO: this entire section uses meteor
+
+    // There could be an existing user with an invite still pending (not activated).
+    // We create a new account only if there's no pending invite.
+    if (!invitedUser) {
+      // The user does not already exist, we need to create a new account
+      userId = MeteorAccounts.createUser({
```

Comment:
@aldeed how do we go about creating `users` and `MeteorAccounts` without using `meteor`?

Reply:
For now, yes. `const userId = await context.createUser(/* same options */);`

This is one of the big sticky questions. I added `createUser` on context at some point in hopes of that being a solution, but I'm not sure. In the long run, we should be calling out to the separate identity service, asking it to create a new user. But we aren't quite there yet and currently IDP lives in this same codebase. `createUser` just calls through to Meteor's Accounts.createUser, so it's not so much as solution as just centralizing that Meteor code. Eventually we'll have to remove that Meteor dependency, too, but for this PR, `context.createUser` should work.